app/middlewares/users/validations.js

```diff
@@ -97,7 +97,12 @@ const isAuthenticated = async (req, res, next) => {
     return res.status(statusCodes.unauthorized).send('The token was not given');
   }
 
-  const user = await obtainOneUser({ where: { email: verifyToken(token).email } });
+  const user = await getUserObjectByToken(token, res);
+
+  if (user === 'error') {
```

Comment:
This is not a good practice, because your changing the object type. One thing you can do is that in case of error, `getUserObjectByToken` returns an object with an error attribute, like this `{ errors: [ 'One error', 'Another error' ] }`

So here instead of checking if `user === 'error'` you can do `if (user.errors.length)`